django/urls/resolvers.py

```diff
@@ -152,12 +117,98 @@ def _check_pattern_startswith_slash(self):
             return []
 
 
-class RegexURLPattern(LocaleRegexProvider):
-    def __init__(self, regex, callback, default_args=None, name=None):
-        LocaleRegexProvider.__init__(self, regex)
+class RegexURLMixin:
+    def compile(self):
+        try:
+            return re.compile(str(self.pattern)), {}
+        except re.error as e:
+            raise ImproperlyConfigured(
+                '"%s" is not a valid regular expression: %s' % (self.pattern, e)
+            )
+
+
+_PATH_PARAMETER_COMPONENT_RE = re.compile(
+    '<(?:(?P<converter>[^:]+):)?(?P<parameter>\w+)>'
+)
+
+
+def _route_to_regex(route):
+    parts = ['^']
+    converters = {}
+    while True:
+        match = _PATH_PARAMETER_COMPONENT_RE.search(route)
+        if not match:
+            parts.append(re.escape(route))
+            break
+
+        parts.append(re.escape(route[:match.start()]))
+        route = route[match.end():]
+
+        parameter = match.group('parameter')
+        if not parameter.isidentifier():
+            msg = "Parameter name {!r} is not a valid identifier.".format(
+                parameter
+            )
+            raise ImproperlyConfigured(msg)
+        raw_converter = match.group('converter')
+        if raw_converter is None:
+            # If no converter is specified, the default is ``string``.
+            raw_converter = 'string'
```

Comment:
Using `register_converter`, it's possible (but unlikely) somebody registers a different "string" converter. Should we enforce using the default converter, or the converter most recently registered for strings?